### PR TITLE
refactor(theme/typography): update typography with DS

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
## Description

Update typography theme configuration to be in sync with the latest DS updates in Figma

### CODE REVIEW

**Comment by minimalsm** [comment]
@coderabbitai review

**Comment by TylerAPfledderer** [comment]
- [ ] @TylerAPfledderer needs to review Figma threads for additional changes or reversions

**Comment by TylerAPfledderer** [comment]
@pettinarip if you are good here, would like to go ahead and have @nloureiro double-check these.

- [UI Review Changeset](https://www.chromatic.com/review?appId=63b7ea99632763723c7f4d6b&number=12853&type=linked&view=changes)
- [Preview Heading component story](https://63b7ea99632763723c7f4d6b-epfyvmtkek.chromatic.com/?path=/story/atoms-typography-heading--heading)
- [Preview Text component stories](https://63b7ea99632763723c7f4d6b-epfyvmtkek.chromatic.com/?path=/story/atoms-typography-text--normal)

**Comment by pettinarip** [comment]
> @pettinarip if you are good here, would like to go ahead and have @nloureiro double-check these.
> 
> * [UI Review Changeset](https://www.chromatic.com/review?appId=63b7ea99632763723c7f4d6b&number=12853&type=linked&view=changes)
> * [Preview Heading component story](https://63b7ea99632763723c7f4d6b-epfyvmtkek.chromatic.com/?path=/story/atoms-typography-heading--heading)
> * [Preview Text component stories](https://63b7ea99632763723c7f4d6b-epfyvmtkek.chromatic.com/?path=/story/atoms-typography-text--normal)

Interesting. Didn't know that there were these many differences xD

Sure. I'll review this later today. Thanks for the PR

**Comment by nloureiro** [comment]
@TylerAPfledderer I can´t really tell about the code changes, but I did a run on the dev preview, and everything seems in place! I only found one issue. 

Other than that, LFG!!! Thank you!


only issue: this fake h1 should be 1rem and not 1.25.
Screen Shot 2024-05-02 10 29 53 AM

**Comment by TylerAPfledderer** [comment]
@nloureiro that adjustment has been made

**Review by nloureiro — APPROVED** [review]

**Comment by nloureiro** [comment]
> @nloureiro that adjustment has been made

awesome!  Thank you!

**Review by wackerow — APPROVED** [review]

File: src/@chakra-ui/components/Heading.ts [hunk 1]
```diff
@@ -1,38 +1,44 @@
...
-      [key]: defineStyle({
-        ...value,
-        lineHeight: lineHeightScale[key],
-      }),
-    }
-  },
-  {
-    ...defaultSizes,
+const fontSizeScale: SIZE_SCALE = {
+  "3xl": ["5xl", null, null, "6xl"],
```
Comment (pettinarip) [inline src/@chakra-ui/components/Heading.ts:22]: This size is not documented in the figma file, right?
image
We have 7 sizes in the code but 6 in figma. Should we have this size in the code? maybe not. cc @nloureiro

File: src/@chakra-ui/styles.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ const styles = {
     body: {
       bg: "background.base",
       lineHeight: "base",
-      fontSize: "md",
+      fontSize: ["sm", null, null, "md"],
```
Comment (pettinarip) [inline src/@chakra-ui/styles.ts:15]: wooha! big change going on here 🚀

**Review by pettinarip — APPROVED** [review]
@TylerAPfledderer looks good on my side. Big impact going on in this PR.

Left a question about an extra heading size, I would like to have consistency with the figma file if possible.

File: src/@chakra-ui/components/Heading.ts (hunk 1 above)
Comment (TylerAPfledderer) [reply src/@chakra-ui/components/Heading.ts:22]: Whoops! No that should not be there. I'll have it removed.

Comment (TylerAPfledderer) [reply src/@chakra-ui/components/Heading.ts]: @pettinarip this is now updated, removing the `3xl` token completely.

**Pull request merged**